sympy/integrals/transforms.py

```diff
@@ -1137,6 +1190,607 @@ def sbs(expr):
+    (besseli(n, a*t),
+     a**n/(sqrt(s**2-a**2)*(s+sqrt(s**2-a**2))**n),
+     And(a>0, n>-1), Abs(a), dco),
+    # 8.12
+    (t**b*besseli(n, a*t),
+     2**n/sqrt(pi)*gamma(n+S.Half)*a**n*(s**2-a**2)**(-n-S.Half),
+     And(And(a>0, n>-S.Half), Eq(b, n)), Abs(a), dco),
+    # 8.13
+    (t**b*besseli(n, a*t),
+     2**(n+1)/sqrt(pi)*gamma(n+S(3)/2)*a**n*s*(s**2-a**2)**(-n-S(3)/2),
+     And(And(a>0, n>-1), Eq(b, n+1)), Abs(a), dco),
+    # 8.15, 8.16
+    (t**(b)*besseli(n, 2*sqrt(a*t)),
+     a**(n/2)*s**(-n-1)*exp(a/s),
+     And(And(a>0, n>-1), Eq(b, n*S.Half)), S.Zero, dco),
+    # 8.17
+    (bessely(0, a*t),
+     -2/pi*asinh(s/a)/sqrt(s**2+a**2),
+     a>0, S.Zero, dco),
+    # 8.18
+    (besselk(0, a*t),
+     (log(s+sqrt(s**2-a**2)))/(sqrt(s**2-a**2)),
+     a>0, Abs(a), dco)
+    ]
+    return laplace_transform_rules
+
+def _laplace_cr(f, a, c, **hints):
+    """
+    Helper function that will return `(f, a, c)` unless `**hints` contains
+    `noconds=True`, in which case it will only return `f`.
+    """
+    conds = not hints.get('noconds', False)
+    if conds:
+        return f, a, c
+    else:
+        return f
+
+def _laplace_rule_timescale(f, t, s, doit=True, **hints):
+    """
+    This function tries to apply the time-scaling rule of the Laplace
+    transform and returns `None` if it cannot do it.
+    """
+    _simplify = hints.pop('simplify', True)
+    b = Wild('b', exclude=[t])
+    g = WildFunction('g', nargs=1)
+    k, func = f.as_independent(t, as_Add=False)
+    ma1 = func.match(g)
+    if ma1:
+        arg = ma1[g].args[0].collect(t)
+        ma2 = arg.match(b*t)
+        if ma2 and ma2[b]>0:
+            debug('_laplace_apply_rules match:')
+            debug('      f:    %s ( %s, %s )'%(f, ma1, ma2))
+            debug('      rule: amplitude and time scaling (1.1, 1.2)')
+            if ma2[b]==1:
+                if doit==True and not any(func.has(t) for func
+                                          in ma1[g].atoms(AppliedUndef)):
+                    return k*_laplace_transform(ma1[g].func(t), t, s,
+                                                simplify=_simplify)
+                else:
+                    return k*LaplaceTransform(ma1[g].func(t), t, s, **hints)
+            else:
+                L = _laplace_apply_rules(ma1[g].func(t), t, s/ma2[b],
+                                         doit=doit, **hints)
+                try:
+                    r, p, c = L
+                    return (k/ma2[b]*r, p, c)
+                except TypeError:
+                    return k/ma2[b]*L
+    return None
+
+def _laplace_rule_heaviside(f, t, s, doit=True, **hints):
+    """
+    This function tries to transform a product containing the `Heaviside`
+    function and returns `None` if it cannot do it.
+    """
+    hints.pop('simplify', True)
+    a = Wild('a', exclude=[t])
+    b = Wild('b', exclude=[t])
+    y = Wild('y')
+    g = WildFunction('g', nargs=1)
+    k, func = f.as_independent(t, as_Add=False)
```

Comment:
It looks like most (all?) rules call `as_independent` like this. Maybe that should be done before the rule function is called.

Reply:
Presently it is for all rules, but this cannot be guaranteed to be true for rules that will come later, or that users can supply themselves according to what @asmeurer suggests below, so I will just leave it as is.